src/transformState.js

```diff
@@ -5,7 +5,26 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action['extraData']);
+        break;
+
+      case 'removeProperties':
+        for (const key of action['keysToRemove']) {
+          delete state[key];
+        }
+        break;
+
+      default:
+        for (const key in state) {
+          delete state[key];
+        }
```

Comment:
it can be like this for this task, yet if you would call the function like this:
`transformState({}, {type: 'unknown type'})` - it will trigger the default case of your switch.

So it's better to explicitly cover all the known cases and for default - provide an explanatory fallback, like
```
throw new Error(`${action} is not supported`)
```